src/punctuated.rs

```diff
@@ -976,6 +1066,36 @@ where
     }
 }
 
+#[cfg(feature = "clone-impls")]
+#[cfg_attr(doc_cfg, doc(cfg(feature = "clone-impls")))]
+impl<'a, T: 'a, P: 'a> Pair<&'a T, &'a P>
+where
+    T: Clone,
+    P: Clone,
+{
+    fn cloned(&self) -> Pair<T, P> {
```

Comment:
This is the wrong signature for a `cloned` function. It would be unusual for someone to be dealing with a `&Pair<&T, &P>`. They would either have a `&Pair<T, P>` or a `Pair<&T, &P>`.

Compare to https://doc.rust-lang.org/1.65.0/std/option/enum.Option.html#method.cloned or https://doc.rust-lang.org/1.65.0/std/result/enum.Result.html#method.cloned or https://doc.rust-lang.org/1.65.0/std/ops/enum.Bound.html#method.cloned.